src/securemul.rs

```diff
@@ -241,8 +260,63 @@ mod tests {
         assert_eq!(Fp31::from(24_u128), validate_and_reconstruct(result_shares));
     }
 
+    /// This test ensures that many secure multiplications can run concurrently as long as
+    /// they all have unique id associated with it. Basically it validates
+    /// `TestHelper`'s ability to distinguish messages of the same type sent towards helpers
+    /// executing multiple same type protocols
+    #[tokio::test]
+    #[allow(clippy::cast_possible_truncation)]
+    pub async fn concurrent_mul() {
+        let ring = helpers::ring::mock::make_three();
+        let participants = crate::prss::test::make_three();
+        let context = make_context(&ring, &participants);
+        let mut rand = StepRng::new(1, 1);
+        let a = share(Fp31::from(4_u128), &mut rand);
+        let b = share(Fp31::from(3_u128), &mut rand);
+
+        let mut multiplications = Vec::new();
+        for i in 1..10_u128 {
+            // there is something weird going on the compiler's side. I don't see why we need
+            // to use async move as `i` is Copy + Clone, but compiler complains about it not living
+            // long enough
+            let ctx = &context;
+            let f = async move {
+                let h1_future = SecureMul {
+                    step: TestStep::Mul1(i as u8),
+                    index: i,
+                    a_share: a[0],
+                    b_share: b[0],
+                }
+                .execute(&ctx[0]);
+                let h2_future = SecureMul {
+                    step: TestStep::Mul1(i as u8),
+                    index: i,
+                    a_share: a[1],
+                    b_share: b[1],
+                }
+                .execute(&ctx[1]);
+                let h3_future = SecureMul {
+                    step: TestStep::Mul1(i as u8),
+                    index: i,
+                    a_share: a[2],
+                    b_share: b[2],
```

Comment:
I thought that we decided to add a SecureMul::new() method for this?  I note that you are placing arguments in a different order in this test and the other.  FWIW, I think that this ordering makes sense.

Reply:
fair point, I was trying to make a change with as few changed lines as possible to make review easier, but I could fix this too.